src/Parser.js

```diff
@@ -33,19 +36,41 @@ class Parser {
   }
 
   registerExtension(ext, parser) {
-    if (!ext.startsWith('.')) {
+    const type = toType(ext);
+
+    if (type === 'regexp' || isRegexp.test(ext)) {
```

Comment:
Is the type check alone not sufficient?

Reply:
Do you mean the expression cannot tell if ext is a regex or not? Could you give me some example?

Reply:
I meant are both the `type === 'regexp' and `isRegexp.test(ext)` checks necessary. Seems they're both checking if it's a regex.

```js
if (type === 'regexp') {
     return key
} else if (ext === key) {
// ...
```

Reply:
Oh I see, I just wanna deal with this situation:  
```javascript
bundler.addAssetType('/ext/', require.resolve('./MyAsset'));
```
In some cases type name may generated by a function:
```javascript
() => `/${['jpg', 'jpeg'].map(type => `(${type})`).join('|')}/`
```
So they donot need to cover string to RegExp.